### PR TITLE
Problem: We need to add filter option for GET documents REST API call.

### PR DESCRIPTION
**Pull request by jana-rapava**
Solution: Add it.

Signed-off-by: Jana Rapava <janarapava@eaton.com>

### CODE REVIEW

File: src/41_secw-rest.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <!-- Read document collection-->
     <mapping>
       <target>rest_documents_GET@libfty_security_wallet_rest</target>
-      <url>^/api/v1/admin/security-wallet/portfolios/([a-zA-Z-]+)/documents$</url>
+      <url>^/api/v1/admin/security-wallet/portfolios/([a-zA-Z-]+)/documents.*$</url>
```
Comment (perrettecl) [inline src/41_secw-rest.xml]: Are you sure about the need to change the regex ? If it's needed to add the parameters (documents?usage=usage) we need to make the regex more explicit but I'm not sure it's needed as we could use <args></args>

File: src/rest_documents_GET.ecpp [hunk 2]
```diff
@@ -32,11 +32,12 @@
 #include <dirent.h>
 #include <unistd.h>
 
+#include <cxxtools/jsondeserializer.h>
 #include <tnt/httperror.h>
 
 #include "helpers.h"
 
-#include "fty_security_wallet_library.h"
+#include "fty_security_wallet.h"
```
Comment (perrettecl) [inline src/rest_documents_GET.ecpp]: This link is not necessary in my opinion. The previous include was more explicit.

**Review by perrettecl — CHANGES_REQUESTED** [review]

File: src/41_secw-rest.xml (hunk 1 above)
Comment (jana-rapava) [reply src/41_secw-rest.xml]: It won't match the REST API call without changing the regex, but we can make it more explicit. What would we like? '?usage=[a-zA-z-_]+'
Please take into account that this regex must match empty string as well, if no params are provided

File: src/rest_documents_GET.ecpp (hunk 2 above)
Comment (jana-rapava) [reply src/rest_documents_GET.ecpp]: AFAIK <library_name>_library.h include file is considered internal/helper by zproject and outside libraries should link with <library_name>.h. @EldoreiJK , what do you think?

File: src/41_secw-rest.xml (hunk 1 above)
Comment (perrettecl) [reply src/41_secw-rest.xml]: so we could have have something like documents($|?.+$)

File: src/rest_documents_GET.ecpp (hunk 2 above)
Comment (perrettecl) [reply src/rest_documents_GET.ecpp]: I would even say that we should not use those one it this particular case but "secw_producer_accessor.h" and "secw_exception.h"

File: src/41_secw-rest.xml (hunk 1 above)
Comment (jana-rapava) [reply src/41_secw-rest.xml]: Fixed

File: src/rest_documents_GET.ecpp (hunk 2 above)
Comment (jana-rapava) [reply src/rest_documents_GET.ecpp]: Fixed

**Review by perrettecl — APPROVED** [review]

**Pull request merged**